aten/src/ATen/StorageImpl.h

```diff
@@ -74,42 +70,72 @@ struct AT_API StorageImpl : public Retainable {
 
   template <typename T>
   inline T* unsafe_data() const {
-    return static_cast<T*>(this->data_ptr.get());
+    return static_cast<T*>(this->data_ptr_.get());
   }
 
   void release_resources() {
-    if (finalizer) {
-      (*finalizer)();
+    if (finalizer_) {
+      (*finalizer_)();
     }
-    finalizer = nullptr;
-    data_ptr.clear();
+    finalizer_ = nullptr;
+    data_ptr_.clear();
   }
 
   void operator=(const StorageImpl&) = delete;
 
   virtual size_t elementSize() const {
-    return at::elementSize(scalar_type);
+    return at::elementSize(scalar_type_);
   }
 
   Type& type();
 
-  //TODO: Rename to size() and size to size_
-  size_t get_size() const { 
-    return size;
+  // TODO: Rename to size() and size to size_
+  ptrdiff_t size() const {
+    return size_;
+  };
+  void set_size(ptrdiff_t size) {
+    size_ = size;
+  };
+  bool resizable() const {
+    return resizable_;
+  };
+  at::DataPtr& data_ptr() {
+    return data_ptr_;
+  };
+  void set_data_ptr(at::DataPtr&& data_ptr) {
+    data_ptr_ = std::move(data_ptr);
   };
   void* data() {
-    return data_ptr.get();
+    return data_ptr_.get();
   };
   const void* data() const {
-    return data_ptr.get();
+    return data_ptr_.get();
+  };
+  at::Allocator* allocator() {
+    return allocator_;
+  };
+  at::ScalarType& scalar_type() {
+    return scalar_type_;
+  };
+  const at::Allocator* allocator() const {
+    return allocator_;
   };
-
   int getDevice() const {
-    return data_ptr.device().index();
+    return data_ptr_.device().index();
   }
-  void set_resizable(bool resizable_) {
-    resizable = resizable_;
+  void set_resizable(bool resizable) {
+    resizable_ = resizable;
   }
+
+ private:
+  at::ScalarType scalar_type_;
+  at::DataPtr data_ptr_;
+  ptrdiff_t size_;
+  bool resizable_;
+
+ public:
+  at::Allocator* allocator_;
```

Comment:
You'll make these private soon?

Reply:
Yes, in an upcoming PR to keep this one small.